### DISCUSSION TITLE
RTU Modbus sniffer

### DISCUSSION THREAD

**Discussion by MarkoBursic** [sanny32/OpenModScan#19] [Q&A] [answered]
Can this application work also as Modbus sniffer? I would like to listen the conversation between two devices RTU Master<->Slave, so I don't want my PC acts as Master, just listen and decode what they send.

**Comment by sanny32** [comment, marked as answer]
No, there is no such functionality.

**Comment by g6094199** [comment]
to bring this up again.

since we have the show traffic option it would be nice to have the ability to show the bus data as slave.

backougrund ist: we have a modbus to LoRaWAN adapter which acts as master (asks for specific content on the bus) but ALSO transmitts EVERY data that is transfered from other masters/slaves on the bus.

**Reply by MarkoBursic** [reply]
You can try with Wireshark + additional feature [https://github.com/jzhvymetal/WiresharkSerialAdapter](url). There is also a practical manual from  ABB: [https://library.e.abb.com/public/053acd5e47a04d938e15537a5d253e74/Technical_Note_179_WiresharkForModbusRTU.pdf?x-sign=zHGIPySTjLhl7dpEkso9HT4DePnDDowfPgQ6XIJ7IXlmfbUxtfYU5tLt4aiRTA3M](url). 

Similarly you an also snif BACnet MS/TP: [https://library.e.abb.com/public/2b037b07870b4af3831e88afe0bcf1a2/LVD-EOTKN123U-EN_WiresharkforBACnetMSTPtrafficcapture_REVA.pdf](url)

**Comment by sanny32** [comment]
Application show traffic between modbus client (master) and server (slave). There is no option to sniff other traffic.

**Reply by g6094199** [reply]
it would be usefull. 
can you add it to your the featurerequest list?

**Reply by sanny32** [reply]
There is no plan to do this.
